Common/Helpers/AzureTableStorageManager.cs

```diff
@@ -10,37 +11,39 @@
 
 namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Helpers
 {
-    public class AzureTableStorageHelper : IAzureTableStorageHelper
+    public class AzureTableStorageManager : IAzureTableStorageManager
     {
-        private readonly string _storageConnectionString;
-        private readonly string _tableName;
-        private CloudTable _table;
+        private readonly ICloudTableProvider _cloudTableProvider;
 
-        public AzureTableStorageHelper(string storageConnectionString, string tableName)
+        public AzureTableStorageManager(string storageConnectionString, string tableName)
         {
-            _storageConnectionString = storageConnectionString;
-            _tableName = tableName;
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(_storageConnectionString);
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(storageConnectionString);
             CloudTableClient tableClient = storageAccount.CreateCloudTableClient();
-            _table = tableClient.GetTableReference(_tableName);
+            _cloudTableProvider = new CloudTableProvider(tableClient, tableName);
         }
 
-        public async Task<CloudTable> GetTableAsync()
+        public TableResult Execute(TableOperation tableOperation)
         {
-            await _table.CreateIfNotExistsAsync();
-            return _table;
+            CloudTable table = _cloudTableProvider.GetCloudTable();
```

Comment:
@handeadiguzel Any construction logic that potentially requires reaching out to another service / remote calls should be avoided. This can cause request startup itself to fail if the dependency isn't available (Constructor threw an exception, etc.) where the request didn't quite need to make a call to the storage provider. Plus it makes construction success based on an external resource which isn't a recommended pattern.